apps/expert/lib/expert/port.ex

```diff
@@ -41,7 +41,16 @@ defmodule Expert.Port do
     # managed programs.
     shell = System.get_env("SHELL")
 
-    {path, 0} = System.cmd(shell, ["-i", "-l", "-c", "cd #{root_path} && echo $PATH"])
+    # Ideally, it should contain the path to shell (e.g. `/usr/bin/fish`),
+    # but it might contain only the name of the shell (e.g. `fish`).
+    is_fish? = String.contains?(shell, "fish")
+
+    # Fish uses space-separated PATH, so we use the built-in `string join` command
+    # to join the entries with colons and have a standard colon-separated PATH output
+    # as in bash, which is expected by `:os.find_executable/2`.
+    path_command = if is_fish?, do: "string join ':' $PATH", else: "echo $PATH"
```

Comment:
nit: we prefer something like `fish_shell?`, `is_x?` mixes both naming conventions for guards and booleans